src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp

```diff
@@ -180,93 +167,37 @@ class boss_mother_shahraz : public CreatureScript
             switch (eventId)
             {
                 case EVENT_RANDOM_BEAM:
-                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    {
-                        switch (CurrentBeam)
-                        {
-                            case SINISTER_BEAM:
-                                DoCast(target, SPELL_BEAM_SINISTER);
-                                break;
-                            case VILE_BEAM:
-                                DoCast(target, SPELL_BEAM_VILE);
-                                break;
-                            case WICKED_BEAM:
-                                DoCast(target, SPELL_BEAM_WICKED);
-                                break;
-                            case SINFUL_BEAM:
-                                DoCast(target, SPELL_BEAM_SINFUL);
-                                break;
-                            default:
-                                break;
-                        }
-
-                        ++BeamCount;
-                        uint32 Beam = CurrentBeam;
-                        if (BeamCount > 3)
-                            while (CurrentBeam == Beam)
-                                CurrentBeam = urand(0, 3);
-                    }
-                    events.ScheduleEvent(EVENT_RANDOM_BEAM, 9000);
+                    DoCastSelf(BeamTriggers[urand(0, 3)]);
```

Comment:
I think `Trinity::Containers::SelectRandomContainerElement(BeamTriggers)` can be used here.